app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/ThrottlingRate.tsx

```diff
@@ -0,0 +1,220 @@
+import { logMessage } from "@lib/logger";
+
+const THROTTLE_EXPIRY = {
+  weeks: "weeks",
+  permanent: "permanent",
+  default: "default",
+} as const;
+type ObjectValues<T> = T[keyof T];
+export type ThrottleExpiry = ObjectValues<typeof THROTTLE_EXPIRY>;
+
+export const ThrottlingRate = ({ formId }: { formId: string }) => {
+  const {
+    t,
+    i18n: { language },
+  } = useTranslation("admin-settings");
+  const hasHydrated = useRehydrate();
+  const [loadedSetting, setLoadedSetting] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  const [weeks, setWeeks] = useState("");
+  const [weeksDisabled, setWeeksDisabled] = useState(false);
+  const [permanent, setPermanent] = useState(false);
+
+  const [success, setSuccess] = useState<ThrottleExpiry | "">("");
+
+  const dateFormat: Intl.DateTimeFormatOptions = {
+    timeZone: "America/New_York",
+    year: "numeric",
+    month: "2-digit",
+    day: "2-digit",
+    hour: "2-digit",
+    minute: "2-digit",
+    hour12: false,
+  };
+
+  const formatDate = (weeks: number, language: string) => {
+    try {
+      const weeksInSeconds = getWeeksInSeconds(weeks);
+      const futureTimeInSeconds = Date.now() + weeksInSeconds;
+      const dateInUTC = new Date(futureTimeInSeconds).toUTCString();
+      const { month, day, year } = formClosingDateEst(dateInUTC, language, dateFormat);
+
+      if (day && month && year) {
+        return `${year}-${month}-${day}`;
+      }
+
+      throw new Error("Failed to parse out day, month or year.");
+    } catch (error) {
+      logMessage.info("Unable to parse throttling in weeks", weeks);
+      return null;
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setSubmitting(true);
+    try {
+      if (permanent) {
+        await setPermanentThrottling(formId);
+        setSuccess(THROTTLE_EXPIRY.permanent);
+        return;
+      }
+
+      if (weeks) {
+        await setThrottlingExpiry(formId, Number(weeks));
+        setSuccess(THROTTLE_EXPIRY.weeks);
+        return;
+      }
+
+      // Reset throttling back to default
+      await resetThrottling(formId);
+      setSuccess(THROTTLE_EXPIRY.default);
+    } catch (error) {
+      toast.error(t("throttling.error"));
+    } finally {
+      setSubmitting(false);
+    }
+  };
+
+  useEffect(() => {
+    const getThrottlingSetting = async () => {
+      try {
+        const { rate, expires } = await getThrottling(formId);
+        if (rate && expires < 0) {
+          setWeeksDisabled(true);
+          setPermanent(true);
+        }
+        if (rate && expires > 0) {
+          const secondsAsWeeks = getSecondsInWeeks(expires - Date.now());
+          setWeeks(String(secondsAsWeeks));
+        }
+      } catch (error) {
+        toast.error(t("throttling.error"));
+      } finally {
+        setLoadedSetting(true);
+      }
+    };
+    setLoadedSetting(false);
+    getThrottlingSetting();
+  }, [formId, t]);
```

Comment:
Let's set the translation outside of the useEffect so we don't have `t` as a dep --- I know we have done this before but we should try to avoid.